README.md

```diff
@@ -84,40 +84,40 @@ delegates as much as possible to the NFA engine.
 
 ## Theory
 
-**(This section is written in a somewhat informal style; I hope to
-expand on it)**
-
-The fundamental idea is that it's a backtracking VM like PCRE, but as
-much as possible it delegates to an "inner" RE engine like RE2 (in
-this case, the Rust one). For the sublanguage not using fancy
-features, the library becomes a thin wrapper.
-
-Otherwise, you do an analysis to figure out what you can delegate and
-what you have to backtrack. I was thinking it might be tricky, but
-it's actually quite simple. The first phase, you just label each
-subexpression as "hard" (groups that get referenced in a backref,
-look-around, etc), and bubble that up. You also do a little extra
-analysis, mostly determining whether an expression has constant match
-length, and the minimum length.
-
-The second phase is top down, and you carry a context, also a boolean
-indicating whether it's "hard" or not. Intuitively, a hard context is
-one in which the match length will affect future backtracking.
-
-If the subexpression is easy and the context is easy, generate an
-instruction in the VM that delegates to the inner NFA implementation.
-Otherwise, generate VM code as in a backtracking engine. Most
-expression nodes are pretty straightforward; the only interesting case
-is concat (a sequence of subexpressions).
-
-Even that one is not terribly complex. First, determine a prefix of
-easy nodes of constant match length (this won't affect backtracking,
-so safe to delegate to NFA). Then, if your context is easy, determine
-a suffix of easy nodes. Both of these delegate to NFA. For the ones in
-between, recursively compile. In an easy context, the last of these
-also gets an easy context; everything else is generated in a hard
-context. So, conceptually, hard context flows from right to left, and
-from parents to children.
+The core concept behind this library is to implement a backtracking virtual machine (VM) for regular expression matching, similar to PCRE.
+However, whenever possible, this VM delegates work to an underlying regular expression engine - the Rust regex crate - which does not otherwise support "fancy" features like lookarounds and backreferences.
```

Comment:
It would be good to mention the "why" here, i.e. because the regex crate has runtime linear to input length.